ledger/src/blockstore_db.rs

```diff
@@ -1271,6 +1289,47 @@ impl<C> LedgerColumn<C>
 where
     C: TypedColumn + ColumnName,
 {
+    #[allow(clippy::needless_range_loop)]
+    pub fn multi_get(&self, keys: Vec<C::Index>) -> Vec<Result<Option<C::Type>>> {
+        let mut rocks_keys: Vec<Vec<u8>> = vec![];
+        for key in keys {
+            rocks_keys.push(C::key(key));
+        }
```

Comment:
Rust complains about mismatched types when I did this:

```
1294 |         let mut rocks_keys: Vec<Vec<u8>> = keys.iter().map(|k| C::key(k)).collect();
     |                                                                ------ ^ expected associated type, found reference
     |                                                                |
     |                                                                arguments to this function are incorrect
```

When I changed to *k instead, then it complains move occurs and C::Index does not implement Copy trait:

```
error[E0507]: cannot move out of `*k` which is behind a shared reference
    --> ledger/src/blockstore_db.rs:1294:71
     |
1294 |         let mut rocks_keys: Vec<Vec<u8>> = keys.iter().map(|k| C::key(*k)).collect();
     |                                                                       ^^ move occurs because `*k` has type `<C as Column>::Index`, which does not implement the `Copy` trait
```

Do you happen to know how I should fix this?

Reply:
Yeah, the `key()` function defined in `Column` requires the type and not a ref:
```rust
pub trait Column {
    type Index;
    ...
    fn key(index: Self::Index) -> Vec<u8>;
```
But, in this case, `multi_get()` takes ownership of `keys`, so you can do `.into_iter()` to consume `keys`:
```rust
let rocks_keys: Vec<_> = keys.into_iter().map(|key| C::key(key)).collect();
```